graylog2-server/src/main/java/org/graylog2/indexer/IndexMapping.java

```diff
@@ -18,48 +18,91 @@
 
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
+import org.elasticsearch.action.ActionFuture;
 import org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequest;
-import org.elasticsearch.action.admin.indices.mapping.put.PutMappingRequestBuilder;
+import org.elasticsearch.action.admin.indices.mapping.put.PutMappingResponse;
 import org.elasticsearch.client.Client;
-import org.graylog2.indexer.messages.Messages;
 import org.graylog2.plugin.Tools;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
 import java.io.Serializable;
 import java.util.List;
 import java.util.Map;
 
+import static com.google.common.base.Preconditions.checkNotNull;
 
 /**
  * Representing the message type mapping in ElasticSearch. This is giving ES more
  * information about what the fields look like and how it should analyze them.
  */
-public class Mapping {
+@Singleton
+public class IndexMapping {
+    public static final String TYPE_MESSAGE = "message";
+    public static final String TYPE_META = "graylog_meta";
```

Comment:
should this maybe be called something more specific?
if we ever decide to store other metadata about the index the name would maybe be a little misleading?
also, since we control the types in the index, i think we can drop the graylog_ prefix.

perhaps `indexrange`  ?